AGENTS.md

```diff
@@ -37,3 +43,62 @@
 ## Configuration & Logging Tips
 - Tune key length and padding with system properties such as `-Dcom.github.simbo1905.srs.BaseRecordStore.MAX_KEY_LENGTH=128` to mirror production limits.
 - Enable `Level.FINEST` logging when investigating disk corruption; redact sensitive keys before sharing traces.
+
+## Documentation Standards
+- Use JEP 467 documentation format with `///` triple-slash comments instead of traditional `/**` javadoc style
+- Place documentation on the line immediately above the element being documented
+- Example: `/// Returns the file path of this record store.` followed by `public Path getFilePath()`
+
+## Logging Standards
+- **CRITICAL**: Never add temporary INFO level logging for debugging purposes
+- Always use appropriate log levels: FINE for normal debugging, FINEST for detailed tracing
+- Use `JulLoggingConfig` system properties to control logging levels: `-Djava.util.logging.config.file=logging.properties`
+- Logging should be permanent and controlled via configuration, not added/removed from code
+- Adding temporary logging and then removing it is considered fraudulent practice
+
+## Critical API Behavior
+
+### maxKeyLength Enforcement
+The `maxKeyLength` parameter is **fundamental and enforced**:
```

Comment:
we should ensure that we do both and Objects.assertNotNull but also that we check the length and throw an IllegalArgumentException if we are not doing so already.